tests/testthat/test-coerce-from.R

```diff
@@ -11,6 +11,8 @@ test_that("Coercion from other date classes into messydt works", {
   expect_equal(as.POSIXlt(messy, mean), POSIXlt)
   expect_equal(as.Date(as_messydate("2010-10-10"), median), date)
   expect_equal(as.Date(as_messydate("2010-10-10"), random), date)
+  expect_error(as.POSIXct(as_messydate("-2012"), min))
+  expect_error(as.POSIXlt(as_messydate("-2012"), min))
```

Comment:
Why should these be errors?

Reply:
This is the case, if I recall correctly, as.POSIXct() does not handle negative dates well. This works, however, with as.Date(). We also have a warning that tells users to use as.Date instead.